openapi/schema.go

```diff
@@ -41,20 +41,26 @@ func (s *Schema) getChildren() map[string]Traversable {
 }
 
 func (s *Schema) setChild(i string, t Traversable) {
-	// TODO: handle this error
-	schema, _ := t.(*Schema)
-	if i == "item" {
-		s.Items = schema
-	} else {
-		s.Properties[i] = schema
+	if schema, ok := t.(*Schema); ok {
+		if i == "item" {
+			s.Items = schema
+		} else {
+			s.Properties[i] = schema
+		}
+		return
 	}
+	panic("(s *Schema) setChild borked")
 }
 
 func (s *Schema) getBasePath() string {
+	if s.parent == nil {
+		return ""
+	}
 	basePath := filepath.Join(s.parent.getBasePath(), filepath.Dir(s.Ref))
 	return basePath
 }
 
-func (s *Schema) getRef() string {
-	return s.Ref
-}
+//func (s *Schema) getRef() string {
+//	panic("(s *Schema) getRef() This should never be called")
+//	return s.Ref
+//}
```

Comment:
This function is still necessary. I can see why you would think otherwise given that node is defined like this:
```
type node struct {
	...
	ref      string
}
```

But that was a mistake and should either be removed or fully refactored.
The unmarshalling logic can't access node's private ref attribute and so wont unmarshal `$ref` keys into it.

If you'd like to remove this permanently, consider making node.ref public and adding a tag like this: `yaml:"$ref"`

You will then also have to remove the Ref attribute from all structs that compose node, such as Schema, because their Ref attribute will shadow node's Ref attribute.

Reply:
have restored it with appropriate implementations in other places. 
Looking for the ref to the content of a Schema to live within the Schema struct itself. Let me know if this makes sense to you?